### PR TITLE
FRESHTEAMS: new dmm1 variation

### PR DESCRIPTION
**Pull request by dusty-qw**
/fresh enables FreshTeams (faster weapon spawns, less ammo) 
/freshpacks toggles pack ammo limits in FreshTeams 
/freshguns toggles weapon sweep ammo limits in FreshTeams 
/freshtime toggles weapon spawn times between 20/15/10 seconds (20 is default) 
/nosweep disables weapon sweeping in dmm1 (does not require FreshTeams)

### CODE REVIEW

File: src/items.c [hunk 1]
```diff
@@ -835,47 +839,79 @@ void weapon_touch()
 
 	if (!strcmp(self->classname, "weapon_nailgun"))
 	{
-		if (leave && ((int)other->s.v.items & IT_NAILGUN))
+		if ((leave || k_nosweep) && ((int)other->s.v.items & IT_NAILGUN))
```
Comment (qqshka) [inline src/items.c:842]: instead of checking k_nosweep set leave to 1 above if k_nosweep != 0

Comment (qqshka) [reply src/items.c:842]: seems I was wrong and it is a bit more complicated and leave flag used in more places than I expected

**Comment by tcsabina** [comment]
@dusty-qw,
We should make such gameplay change visible upon starting the match. When the countdown is visible, I mean.
Could you come up with some way of showing these new functions when they are active?

**Comment by dusty-qw** [comment]
> @dusty-qw, We should make such gameplay change visible upon starting the match. When the countdown is visible, I mean. Could you come up with some way of showing these new functions when they are active?

It tells you during countdown if "freshteams" is enabled or if nosweep is on. Is that what you mean? I didn't put every single config option though because there's quite a few.

**Comment by tcsabina** [comment]
> > @dusty-qw, We should make such gameplay change visible upon starting the match. When the countdown is visible, I mean. Could you come up with some way of showing these new functions when they are active?
> 
> It tells you during countdown if "freshteams" is enabled or if nosweep is on. Is that what you mean? I didn't put every single config option though because there's quite a few.

Yes, that is what I mean.
Perfect!

File: src/commands.c [hunk 2]
```diff
@@ -7311,6 +7328,184 @@ void SetMidairMinHeight()
 }
 
 void W_SetCurrentAmmo();
+
+void ToggleFreshTeams()
+{
+	int k_freshteams = bound(0, cvar("k_freshteams"), 2);
+
+	if (match_in_progress)
```
Comment (qqshka) [inline src/commands.c:7336]: remove match_in_progressm in new commands, it is redundant, is_rules_change_allowed() checks it as well but also with user warnings.

File: src/commands.c [hunk 3]
```diff
@@ -7311,6 +7328,184 @@ void SetMidairMinHeight()
...
+
+	// Can't enable freshteams unless dmm1 is set first
+	if (deathmatch != 1)
+	{
+		G_sprint(self, 2, "FreshTeams requires dmm1\n");
+
+		return;
+	}
+
+	if (!k_freshteams)
```
Comment (qqshka) [inline src/commands.c:7354]: use cvar_toggle_msg() instead or add your own helper.

File: src/commands.c [hunk 4]
```diff
@@ -7311,6 +7328,184 @@ void SetMidairMinHeight()
 }
 
 void W_SetCurrentAmmo();
+
+void ToggleFreshTeams()
+{
+	int k_freshteams = bound(0, cvar("k_freshteams"), 2);
```
Comment (qqshka) [inline src/commands.c:7334]: bound by 2? should be 1 I guess.

**Comment by qqshka** [comment]
Overall it's okay but fix what I mention above, please.
Side note: players cannot see the final settings if fresh teams are enabled, and they can change quite a few, but since it doesn't affect core game modes and it's kind of a testing playground - it is fine for me.

File: src/commands.c (hunk 2 above)
Comment (dusty-qw) [reply src/commands.c]: removed.

File: src/commands.c (hunk 3 above)
Comment (dusty-qw) [reply src/commands.c]: fixed.

File: src/commands.c (hunk 4 above)
Comment (dusty-qw) [reply src/commands.c]: fixed.

**Pull request merged**